### PR TITLE
Change: [CI] Automatically disable asserts when building a stable release

### PR DESCRIPTION
**Pull request by LordAro**
## Motivation / Problem

Remembering to turn off asserts for official releases is often a very last minute thing that has been forgotten entirely in the past on several occasions (most recently, 13.0)

## Description

Use the fact that when we build releases we already have a pre-written .ottdrev available. Parse that and turn off the cmake option as appropriate

## Limitations

Not fully tested:

```
 $ echo -e "foo\tbar\tbaz\tquix\tqneez\t1\tflibble" > .ottdrev
 $ IS_STABLE_TAG=$(if [ "$(cat .ottdrev | cut -f 6 -d$'\t')" = "1" ]; then echo "OFF"; else echo "ON"; fi)
 $ echo $IS_STABLE_TAG
OFF
 $ echo -e "foo\tbar\tbaz\tquix\tqneez\t0\tflibble" > .ottdrev
 $ IS_STABLE_TAG=$(if [ "$(cat .ottdrev | cut -f 6 -d$'\t')" = "1" ]; then echo "OFF"; else echo "ON"; fi)
 $ echo $IS_STABLE_TAG
ON

```

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: cmake/Options.cmake [hunk 1]
```diff
@@ -52,11 +52,32 @@ function(set_options)
...
+                        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+                        ERROR_QUIET
+        )
+        if(NOT ${TAG} STREQUAL "")
+            string(REGEX REPLACE "^[0-9.]+$" "" STABLETAG "${TAG}")
+            if(STABLETAG STREQUAL "")
+                set(DEFAULT_OPTION_USE_ASSERTS OFF)
+            endif()
+        endif()
+    endif()
```
Comment (TrueBrain) [inline cmake/Options.cmake]: I like the approach, I very much dislike the location. Should be outside this function for sure (which is fine in CMake, you just need to do this before this function is called).

**Review by TrueBrain — DISMISSED** [review]
Assuming you finish testing :) Let's be sure it actually works as advertised :D

**Comment by TrueBrain** [comment]
One note: I was thinking about Linux distros that build OpenTTD. Currently, they can just build the tag, and the right asserts yes/no come out. After this PR, they have to flip that switch themselves.

I am not sure what to think about that .. good .. bad .. but it is different. Just wanted to note that :)

**Comment by TrueBrain** [comment]
Another approach to this would be to use a Workflow to create a release, instead of doing it via the release page.

That way, the workflow can check what kind of release this is, and make the correct changes in the form of a commit before making the actual release.

As we recently found out, besides asserts, we also have to set the CMake project version, which we tend to forget.

Another thing that comes to mind when I think along these ways, that the workflow can also make a changelog.txt draft, push that in a PR, and waits for it to be merged.

Dunno .. just thinking out loud here :)

**Comment by LordAro** [comment]
This probably ultimately isn't the right direction to go

**Pull request closed without merging**